feat(avatar): implement user avatar management with social integration

gentamura opened the pull request:
## Summary
Implement user avatar management system that supports both local file uploads and OAuth provider avatars (Google, GitHub).

## Related Issue
N/A

## Changes
- Add avatar upload functionality with public storage
- Implement avatar management policies in Supabase
- Set up system migrations for storage policies
- Import OAuth provider avatars on signup
- Add file validation and size restrictions (1MB limit)
- Create public URL handling for avatars
- Implement cleanup of old avatar files
- Add loading states and error handling
- Update user button and profile components

## Testing
- Tested OAuth avatar import from Google and GitHub
- Verified local image upload with size/format restrictions
- Confirmed public accessibility of uploaded avatars
- Validated old file cleanup functionality
- Tested error states and loading indicators

## Other Information
- Uses public-assets bucket for avatar storage
- Storage policies ensure users can only manage their own avatars
- Supports JPG, PNG, GIF, and WebP formats
- Implements proper file cleanup to prevent storage bloat
- Maintains public URLs for consistent access

gentamura commented on apps/studio.giselles.ai/app/(main)/settings/account/actions.ts:
```diff
@@ -141,3 +154,126 @@ export async function leaveTeam(
+		revalidate: 55 * 60, // Revalidate cache after 55 minutes
+		tags: ["avatar-url"],
+	},
+);
```
I implemented signed URLs to access images from a private bucket. However, this implementation might not be optimal if other team members need to access avatars as well.

A better approach would be to:
1. Create a new public bucket specifically for avatars
2. Store public URLs from Supabase Storage, similar to how we handle social provider avatars
3. Consider implementing custom domain settings for better control and branding

This would improve performance and simplify the implementation while maintaining consistent behavior across different avatar sources.

gentamura commented:
# 📝  Debug notes

## Deleting User Referenced by Email

Due to Supabase Auth account deletion not being synchronized with our database, manual deletion is required. (Without removing the database records, registering with the same email address will result in a duplicate key error.)
The following information serves as a reference for the deletion process.

## User Data Structure and Deletion Steps

### Primary Data
1. `users` table
- User A: email-only registration
- User B: same email with Google authentication

### First Level References
1. `supabase_user_mappings` (referenced by userDbId)
2. `team_memberships` (referenced by userDbId)
3. `agents` (referenced by creatorDbId)
4. `oauth_credentials` (referenced by userId)

### Second Level References
1. `teams` (referenced by team_memberships.teamDbId)
2. `github_integration_settings` (referenced by agents.dbId)
3. `agent_activities` (referenced by agents.dbId)

### Deletion Procedure

Before deletion, check team member count:
```sql
SELECT COUNT(*) 
FROM team_memberships 
WHERE team_db_id = [target_team_id];
```

#### For User A (email-only)
1. Delete from `team_memberships`
2. Delete from `teams` (if last member)
3. Delete from `supabase_user_mappings`
4. Delete from `users`

#### For User B (Google auth)
1. Delete from `team_memberships`
2. Delete from `teams` (if last member)
3. Delete from `agents`
4. Delete from `oauth_credentials`
5. Delete from `supabase_user_mappings`
6. Delete from `users`

gentamura commented:
@shige @toyamarinyon review, please :pray:

shige reviewed (changes requested):
As you mentioned in your own comment, we want the avatar URL to be publicly accessible, so I'd like to ask you to update it to upload to a public bucket.

gentamura commented:
@shige 

I've implemented the requested changes for public avatar access. Here's the current state of the storage policies:

1. Bucket Configuration
- Created policies for the 'public-assets' bucket which is set as a public bucket (named 'public-assets' since 'public' is a reserved keyword)
- Avatars are stored in the 'avatars' subdirectory

2. Storage Policies Implemented
- All storage operations (CREATE, READ, UPDATE, DELETE) require explicit policies, even in public buckets
- Implemented the following policies:
  - Public READ access for all files in the public-assets bucket
  - Authenticated users can:
    - Upload their own avatar (CREATE)
    - Update their own avatar (UPDATE)
    - Delete their own avatar (DELETE)
- File path restrictions ensure users can only manage their own avatars within the specified directory

3. URL Handling
- Using direct public URLs instead of signed URLs
- URLs are in the format: `https://<project>.supabase.co/storage/v1/object/public/public-assets/avatars/<user-id>.<ext>`
- No expiration or authentication required to access the URLs

These changes ensure that:
- Avatar images are publicly accessible
- User permissions are properly restricted
- The implementation aligns with the public access requirement

Please review these changes, particularly:
1. The storage policy implementation
2. The security aspects of public access
3. The file path structure and permissions
4. The URL handling approach

I would greatly appreciate your code review for these changes, particularly focusing on the security implementations and public access handling. Please let me know if you have any questions or if there are areas that need improvement.

shige commented:
Thank you! I will review it tomorrow morning. 🌞

shige reviewed (approved):
Thank you! 🖼️ 

Confirmed: GitHub, Google, and direct image file uploads are all working!
